### DISCUSSION TITLE
Provide me with data structure and arguments operation for this project

### DISCUSSION THREAD

**Discussion by jaipalsin** [bcit-ci/CodeIgniter#6249] [Q&A]
**Codeigniter** here
In the project, I am using Paytm payment gateway for the payment method and I want that when the user adds the item to the cart and goes to the **checkout button**, the user gets redirected to the **user shipping address submission page** where the user has to enter his Name, address, email, etc. have to be submitted. Once completed, the user will be redirected to the **checkout process order page** with the user ID and total amount where he can make the payment. What kind of logic and operation is required for this, I am not able to understand, please help me.

![1](https://github.com/bcit-ci/CodeIgniter/assets/91344021/287652a5-6025-48e7-8466-3a665d723921)

**user shipping address submission**
```
   $this->form_validation->set_rules('name', 'Name', 'required');
   $this->form_validation->set_rules('email', 'Email', 'required|valid_email');
   $this->form_validation->set_rules('password', 'Password', 'required');
   $this->form_validation->set_rules('phone', 'Phone', 'required');
   $this->form_validation->set_rules('address', 'Address', 'required');
```

**checkout process order**

   ```
 $paramList["MID"] = PAYTM_MERCHANT_MID;
    $paramList["ORDER_ID"] = 'id';
    $paramList["CUST_ID"] = 'customer_id';
    $paramList["INDUSTRY_TYPE_ID"] = INDUSTRY_TYPE_ID;
    $paramList["CHANNEL_ID"] = $CHANNEL_ID;
    $paramList["TXN_AMOUNT"] = $item["subtotal"];
    $paramList["WEBSITE"] = PAYTM_MERCHANT_WEBSITE;
   
    $paramList["CALLBACK_URL"] = base_url('Checkout/payment_response');
    $paramList["MSISDN"] = 'xxxxxxxxxx'; //Mobile number of customer
    $paramList["EMAIL"] = 'xxxxxxxxxx'; //Email ID of customer
    $paramList["VERIFIED_BY"] = "MSISDN";
    $paramList["IS_USER_VERIFIED"] = "YES";
   
    $checkSum = $this->paytm->getChecksumFromArray($paramList,PAYTM_MERCHANT_KEY);
    $transaction_url = PAYTM_TXN_URL;
    $data = array('paramList' => $paramList,'transaction_url' => $transaction_url,'checkSum' => $checkSum);
   
    // ,'order_id' => $order_id
    $this->load->view('order_process',$data);
```

**callback_URL**

This is the callback_URL of the checkout process. Here I want if the payment status is successful then I should be shown the complete order information of the user according to the order ID submitted by the user
   ```
 $paytmChecksum = "";
    $paramList = array();
    $isValidChecksum = FALSE;
   
    $paramList = $_POST;
   
    $paytmChecksum = isset($_POST["CHECKSUMHASH"]) ? $_POST["CHECKSUMHASH"] : ""; //Sent by Paytm pg
   
    $isValidChecksum = $this->paytm->verifychecksum_e($paramList, PAYTM_MERCHANT_KEY, $paytmChecksum); //will return TRUE or FALSE string.
     
   
    if($isValidChecksum == "TRUE") {
        if ($_POST["STATUS"] == "TXN_SUCCESS") {

        //Here we can show order information

        }else{
            echo 'Transaction Failed';
        }
    }else{
        echo 'Not Valid Check Sum';
    }

```

**Comment by bunglegrind** [comment]
Session to maintain user status and redirection to move across pages, I suppose. Or if you are a REST fan, model user state as a resource (instead of using the session) and PUT or POST state changes.

Try to clarify your mind with some sequence diagrams.

However, it's not related to codeigniter...

**Reply by jaipalsin** [reply]
HOW CAN I DO THIS CAN YOU  PROVIDE ME WITH SOME CODE FOR THIS

**Reply by bunglegrind** [reply]
Sorry, I haven't code to show you. Just follow the suggestions I wrote above.
